WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java

```diff
@@ -646,12 +648,44 @@ public static boolean isGutenbergEditorEnabled() {
         return true;
     }
 
-    public static boolean shouldAutoEnableGutenbergForTheNewPosts() {
-        return getBoolean(DeletablePrefKey.SHOULD_AUTO_ENABLE_GUTENBERG_FOR_THE_NEW_POSTS, true);
+    public static boolean shouldAutoEnableGutenbergForTheNewPosts(String siteURL) {
+        if (TextUtils.isEmpty(siteURL)) {
+            return false;
+        }
+
+        Set<String> urls;
+        try {
+            urls = prefs().getStringSet(DeletablePrefKey.SHOULD_AUTO_ENABLE_GUTENBERG_FOR_THE_NEW_POSTS.name(), null);
+        } catch (ClassCastException exp) {
+            // It's an old value already stored, means we've already shown the prompt on v12.9
+            return false;
+        }
+        // if the pref was not set before, or doesn't contain the value we can return true
+        return urls == null || !urls.contains(siteURL);
     }
 
-    public static void setGutenbergAutoEnabledForTheNewPosts(boolean enable) {
-        setBoolean(DeletablePrefKey.SHOULD_AUTO_ENABLE_GUTENBERG_FOR_THE_NEW_POSTS, enable);
+    public static void setGutenbergAutoEnabledForTheNewPosts(String siteURL) {
+        if (TextUtils.isEmpty(siteURL)) {
+            return;
+        }
+        Set<String> urls;
+        try {
+            urls = prefs().getStringSet(DeletablePrefKey.SHOULD_AUTO_ENABLE_GUTENBERG_FOR_THE_NEW_POSTS.name(), null);
+        } catch (ClassCastException exp) {
+            // It's an old value, means that we've already shown the prompt
+            // we should not here anyway, since the check returned to not show the prompt
+            return;
+        }
+
+        Set<String> newUrls = new HashSet<>();
+        // re-add the old urls here
+        if (urls != null) {
+            newUrls.addAll(urls);
+        }
+
```

Comment:
It seems we don't add the provided `siteURL` though, is that on purpose? 🤔

Reply:
Fixed in 6e4623b95dd1e9648e9ac9e7deeb66f4c0c33c80

Reply:
Thanks for the fix Danilo!

But now, I'm actually wondering why do we even need the `setGutenbergAutoEnabledForTheNewPosts()` function which keeps flags locally? If no reason at this stage, WDYT about replacing it with a check of the Site Setting for 'emptiness'?

Reply:
Right, there is no need for it anymore.
Addressed in 30cd235fe5b44e1f9939f7d684aaeb967fac0f39